### ISSUE TITLE
`tests_outside_test_module` is triggered when using "complex" `cfg` attributes

### ISSUE THREAD

**Issue by zacknewman** [rust-lang/rust-clippy#16476]
### Summary

Using a "complex" `cfg` attribute like `#[cfg(all(test))]` on a module that contains a function with the `test` attribute causes `tests_outside_test_module` to incorrectly fire. In my case I know the tests in the module will `panic` due to OOM when using a 16-bit architecture since more than `i16::MAX` bytes is needed, so my actual example was using `#[cfg(all(test, not(target_pointer_width = "16")))]`.

### Lint Name

tests_outside_test_module

### Reproducer

I tried this code:

```rust
#[deny(clippy::tests_outside_test_module, reason = "example")]
#[expect(clippy::non_minimal_cfg, reason = "example")]
#[cfg(all(test))]
mod tests {
    #[test]
    fn foo() {}
}
```

I saw this happen:

```bash
error: this function marked with #[test] is outside a #[cfg(test)] module
 --> src/lib.rs:6:5
  |
6 |     fn foo() {}
  |     ^^^^^^^^^^^
  |
  = note: move it to a testing module marked with #[cfg(test)]
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#tests_outside_test_module
  = note: example
note: the lint level is defined here
 --> src/lib.rs:1:8
  |
1 | #[deny(clippy::tests_outside_test_module, reason = "example")]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `example` (lib test) due to 1 previous error
```

I expected to see this happen: no lints to fire

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.93.0
LLVM version: 21.1.8
```

**Comment by tamird** [comment]
This also affects other lints like `expect_used` when configure with `allow-expect-in-tests = true`.
